lib/core/src/lib/notifications/components/notification-history.component.spec.ts

```diff
@@ -182,4 +184,119 @@ describe('Notification History Component', () => {
             });
         }, 45000);
     });
+
+    it('should set unreadNotifications to an empty array when there are no notifications', () => {
+        component.notifications = [];
+        fixture.detectChanges();
+
+        expect(component.unreadNotifications).toEqual([]);
+    });
+
+    it('should set unreadNotifications to an empty array when all notifcations are read', () => {
+        const notifications = [
```

Comment:
You create that almost every test which you added. Could you please avoid duplications?

Reply:
created a mock 'testNotifications' array in beforeEach and modified it depending on the test case.